src/components/change-password/change-form.tsx

```diff
@@ -0,0 +1,171 @@
+        window.fetch(submitUrl, { method: 'POST', body: JSON.stringify(model), headers: { 'Content-Type': 'application/json' } })
+            .then((response) => {
+                const status = response.status;
+                if (status === 0 || (status >= 200 && status < 400)) {
+                    formRef.current!.reset();
+                    postPasswordChangeAction();
+                } else {
+                    response.json().then((res) => {
+                        const errorMessage = res.error.message;
+                        let element;
+
+                        if (status === 400) {
+                            element = newPassInputRef.current!;
+                        } else if (status === 403) {
+                            element = oldPassInputRef.current!;
+                        }
+                        const emptyInputs = {};
+                        invalidateElement(emptyInputs, element!);
+                        setInvalidInputs(emptyInputs);
+                        setErrorMessage(errorMessage);
+                    });
+                }
+            });
+    };
+
+    const postPasswordChangeAction = () => {
+        const action = viewModel.PostPasswordChangeAction;
+
+        if (action === 'ViewAMessage') {
+            const message = viewModel.PostPasswordChangeMessage;
+
+            setSuccessMessage(message);
+        } else if (action === 'RedirectToPage') {
+            const redirectUrl = viewModel.RedirectUrl;
+
+            redirect(redirectUrl);
+        }
+    };
+
+
+    const validateForm = (form: HTMLFormElement) => {
+        setInvalidInputs({});
+        const requiredInputs = form.querySelectorAll('input[data-sf-role=\'required\']');
+        const emptyInputs = {};
+        let isValid = true;
+
+        requiredInputs.forEach((input: any) => {
+            if (!input.value) {
+                invalidateElement(emptyInputs, input);
+                isValid = false;
+            }
+        });
+
+        if (!isValid) {
+            const errorMessage = labels.ValidationRequiredMessage;
+            setErrorMessage(errorMessage);
+            setInvalidInputs(emptyInputs);
+
+            return isValid;
+        }
+
+        const newPassword = newPassInputRef.current!;
+        const repeatPassword = repeatPassInputRef.current!;
+
+        if (isValid && repeatPassword.value !== newPassword.value) {
+            invalidateElement(emptyInputs, repeatPassword);
+            setInvalidInputs(emptyInputs);
+            isValid = false;
+
+            const errorMessage = labels.ValidationMismatchMessage;
+            setErrorMessage(errorMessage);
+        }
+
+        return isValid;
+    };
+
+    return (
+      <form
+        ref={formRef}
+        onSubmit={handleSubmit}
+        {...others}>
+        <h2 className="mb-3">{labels.Header}</h2>
+        { <div data-sf-role="error-message-container"
+          className={classNames('alert alert-danger my-3', {

```

Comment:
I would suggest extracting classNames to a constant before the jsx return. This way the template will be more readable